lib/rack/utf8_sanitizer.rb

```diff
@@ -19,6 +23,25 @@ def call(env)
       @app.call(sanitize(env))
     end
 
+    DEFAULT_STRATEGIES = {
+      replace: proc do |input|
+        input.
+          force_encoding(Encoding::ASCII_8BIT).
+          encode!(Encoding::UTF_8,
+                  invalid: :replace,
+                  undef:   :replace)
+      end,
+      exception: proc do |input|
+        begin
+          input.
+            force_encoding(Encoding::ASCII_8BIT).
+            encode!(Encoding::UTF_8)
+        rescue EncodingError => e
+          raise InvalidEncoding, e.message
```

Comment:
Can't we just reuse `EncodingError`?

Reply:
We could, but then code that depends on the error being raised will have to know the internals of the process. Imagine if in the future two error types are caught or something like that.

Reply:
`encode!` is a public Ruby API, it can't just be changed without any reason. I think we should go with reusing `EncodingError`.

Reply:
Yeah, but my point is that 3rd parties would then need to know that `encode!` is what is throwing the error.
This really isn't a big deal, so I can change it.

Reply:
Yeah, we need to document which error is raised.